cohort-engine/src/main/java/com/ibm/cohort/engine/measure/CDMMeasureEvaluation.java

```diff
@@ -197,6 +205,56 @@ protected static void addDefineEvaluationToReport(MeasureReport report, CDMConte
 			}
 		}
 	}
+	
+	protected static void addParametersToReport(MeasureReport report, Measure measure, Context context, Map<String, Parameter> parameterMap) {
+		Set<String> allParameters = new HashSet<>();
+		
+		// Check for special parameters we handle elsewhere
+		if (context.resolveParameterRef(null, CDMConstants.MEASUREMENT_PERIOD) != null) {
+			allParameters.add(CDMConstants.MEASUREMENT_PERIOD);
+		}
+		
+		if (context.resolveParameterRef(null, CDMConstants.PRODUCT_LINE) != null) {
+			allParameters.add(CDMConstants.PRODUCT_LINE);
+		}
+
+		if (parameterMap != null) {
+			allParameters.addAll(parameterMap.keySet());
+		}
+		
+		List<Extension> parameterExtensions = measure.getExtensionsByUrl(CDMConstants.MEASURE_PARAMETER_URL);
+		for (Extension e : parameterExtensions) {
+			ParameterDefinition parameterDefinition = (ParameterDefinition) e.getValue();
+			allParameters.add(parameterDefinition.getName());
+		}
+		
+		allParameters.forEach(x -> report.addExtension(createParameterExtension(context, x)));
+	}
+	
+	protected static Extension createParameterExtension(Context context, String parameterName) {
+		Object parameterValue = context.resolveParameterRef(null, parameterName);
+
+		Extension innerExtension = new Extension();
+		innerExtension.setUrl("http://ibm.com/fhir/cdm/StructureDefinition/parameter-value");
+		IBaseDatatype fhirParameterValue = MeasureReportParameterHelper.getFhirTypeValue(parameterValue);
+		// Do not create an extension for unsupported types
+		if (fhirParameterValue == null) {
+			return null;
```

Comment:
Very nitpicky, but I would prefer the flow control here to be `if( fhirParameterValue != null )` and then have a single return at the bottom.